src/AutoRest.CSharp/Mgmt/Generation/ResourceGroupExtensionsWriter.cs

```diff
@@ -105,42 +99,34 @@ private void WriteClientMethod(CodeWriter writer, MgmtRestClient restClient, Cli
             // but in parameter invocation, we need to pass a value for `resourceGroupName`
             // instead of `Id.ResourceGroupName` as in normal ResourceContainer or ResourceOperation, we need to pass `resourceGroup.Id.Name`
             var methodParameters = clientMethod.RestClientMethod.Parameters.Skip(1);
-            WriteClientMethod(writer, restClient, clientMethod,
+            WriteExensionClientMethod(writer, restClient, clientMethod,
                 // skip the first parameter, aka the resource group name parameter
                 clientMethod.RestClientMethod.Parameters.Skip(1),
-                BuildPolishedParameterMapping(clientMethod.RestClientMethod),
                 async);
         }
         private void WriteListMethod(CodeWriter writer, MgmtRestClient restClient, PagingMethod pagingMethod, bool async)
         {
-            WriteListMethod(writer, pagingMethod.PagingResponse.ItemType, restClient, pagingMethod,
+            WriteExtensionPagingMethod(writer, pagingMethod.PagingResponse.ItemType, restClient, pagingMethod,
                 // skip the first parameter, aka the resource group name parameter
                 pagingMethod.Method.Parameters.Skip(1),
```

Comment:
In WriuteExtensionPagingMethod, build parameters and remove all pass-through methods